app/scripts/migrations/084.ts

```diff
@@ -0,0 +1,59 @@
+import { cloneDeep } from 'lodash';
+import { isObject } from '@metamask/utils';
+
+export const version = 84;
+
+/**
+ * Ensure that each networkConfigurations object in state.NetworkController.networkConfigurations has an
+ * `id` property which matches the key pointing that object
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (!isObject(state.NetworkController)) {
+    return state;
+  }
+  const { NetworkController }: any = state;
+
+  const {
+    networkConfigurations,
+  }: { networkConfigurations: Record<string, object> } = NetworkController;
+
+  const newNetworkConfigurations: Record<string, object> = {};
+
+  for (const networkConfigurationId in networkConfigurations) {
+    if (
+      Object.prototype.hasOwnProperty.call(
+        networkConfigurations,
+        networkConfigurationId,
+      )
+    ) {
```

Comment:
I'm probably missing something here but why is this check necessary? Won't the `networkConfigurationId` necessarily be a property of `networkConfigurations` as its defined in this the for-in loop?

Reply:
Nit: This check could be skipped if the look was `for (const configurationId of Object.keys(networkConfigurations)) { `instead.